packages/expo-camera/ios/EXCamera/EXCamera.m

```diff
@@ -534,6 +537,12 @@ - (void)record:(NSDictionary *)options resolve:(UMPromiseResolveBlock)resolve re
     UM_WEAKIFY(self);
     dispatch_async(self.sessionQueue, ^{
       UM_STRONGIFY(self);
+      // it is possible that the session has been invalidated at this point
+      // for example, the video codec option is invalid and so this call has already rejected
+      if (!self.isValidRecordingConfig) {
```

Comment:
nit: I think "RecordingConfig" from `isValidRecordingConfig` and "VideoOptions" from `setVideoOptions` are referring to the same thing, so it would be good to pick one way to refer to it and stick to that